experimenter/experimenter/nimbus_ui/views.py

```diff
@@ -605,6 +605,10 @@ class StatusUpdateView(RequestFormMixin, RenderResponseMixin, NimbusExperimentDe
 
     def get_context_data(self, *, form, **kwargs):
         context = super().get_context_data(form=form, **kwargs)
+        origin = self.request.GET.get("origin") or self.request.POST.get("origin")
+        context["render_progress_card_fragment"] = (
+            self.request.headers.get("HX-Request") and origin == "progress_card"
+        )
```

Comment:
Instead of doing this, we can use [`get_template_names()`](https://docs.djangoproject.com/en/5.2/ref/class-based-views/mixins-simple/#django.views.generic.base.TemplateResponseMixin.get_template_names) to determine which template will be rendered.

```python
def get_template_names(self):
    if (
        self.request.headers.get("HX-Request")
        and self.request.GET.get("origin") == "progress_card"
    ):
        return [progress_card_template]

    return [self.template]
```